src/components/AppHeader.vue

```diff
@@ -152,13 +162,46 @@ export default defineComponent({
 							"Successful",
 							`Your DAO App of ID ${this.daoID} is selected.`
 						);
+						if (this.walletStore.address) {
+							isApplicationOpted(this.walletStore.address, this.daoID as number)
+								.then((appIsOptedIn: boolean) => {
+									this.showOptIn = !appIsOptedIn;
+									if (appIsOptedIn) {
+										openSuccessNotificationWithIcon(
+											"You have already Opted-in DAO App"
+										);
+									}
+								})
+								.catch((error) =>
+									openErrorNotificationWithIcon(
+										"Unsuccessful while getting DAO App Opt-in Details",
+										error.message
+									)
+								);
+						}
 					})
 					.catch((error) => {
 						errorMessage(this.key);
 						openErrorNotificationWithIcon("Unsuccessful", error.message);
 					});
 			}
 		},
+		async optIn() {
+			try {
+				await optInDaoApp(
+					this.walletStore.address,
```

Comment:
same here.

Reply:
we are showing optin button only when address and daoID is provided by user, see the condition when we set the `show_opt_in` state to true

Reply:
Added check